File: nomad-samples/admin/py/basic-live-channel-1.1/live_input/update_live_input.py
from constants.project_constants import *
from exceptions.api_exception_handler import *
from helpers.slugify import *
from live_input.live_input_statuses import *
from live_input.live_input_types import *
from live_input.wait_live_input_status import *

import json, requests

def update_live_input(AUTH_TOKEN, ID, NAME, TYPE, SOURCE):
    # Create header for the request
    HEADERS = {
        'Content-Type': 'application/json',
        "Authorization": "Bearer " + AUTH_TOKEN
    }

    # Build the payload BODY
    # Error Object of type set is not JSON serializable
    BODY = { "id": ID }

    if NAME != "":
        BODY["name"] = NAME
        BODY["internalName"] = slugify(NAME)
    if TYPE != "": BODY["type"] = { "id": TYPE }


    # Set the appropriate fields based on the type
    if (TYPE == "RTMP_PUSH"):
        if SOURCE: BODY["sourceCidr"] = SOURCE
    elif (TYPE == "RTMP_PULL" or TYPE == "RTP_PUSH" or TYPE == "URL_PULL"):
        if SOURCE: BODY["sources"] = [{ "url": SOURCE }]

    try:
        RESPONSE = requests.put(ADMIN_URL + "/liveInput",  headers= HEADERS, data= json.dumps(BODY))
    
        # Parse JSON response
        INFO = RESPONSE.json()

        # Wait for the Live Input to be detached if it was just created
        wait_for_live_input_status(AUTH_TOKEN, INFO["id"], LIVE_INPUT_STATUSES["Detached"], 15, 1)

        return INFO
    
    except:
        api_exception_handler(RESPONSE, f"Creating Live Input {NAME} failed")

File: nomad-samples/admin/js/basic-live-channel/live-input/update-live-input.js
import * as prjConstants from "../constants/project-constants.js";
import apiExceptionHandler from "../exceptions/api-exception-handler.js";
import liveInputTypes from "./live-input-types.js";
import liveInputStatuses from "./live-input-statuses.js";
import waitForLiveInputStatus from "./wait-live-input-status.js";
import slugify from "../helpers/slugify.js";


export default async function updateLiveInput(AUTH_TOKEN, ID, NAME, SOURCE, TYPE) {
    // Create header for the request
    const HEADERS = new Headers();
    HEADERS.append("Content-Type", "application/json");
    HEADERS.append("Authorization", `Bearer ${AUTH_TOKEN}`);

    // Build the payload body
    const BODY = {
        id: ID
    };

    if (NAME !== "")
    {
         BODY.name = NAME;
        BODY.internalName = slugify(NAME);
    }
    if (TYPE !== "") BODY.type = { id: liveInputTypes[TYPE] };

    // Set the appropriate fields based on the type
    if (TYPE == "RTMP_PUSH")
    {
        if (SOURCE) BODY["sourceCidr"] = SOURCE
    }
    else if (TYPE === "RTMP_PULL" || TYPE === "RTP_PUSH" || TYPE === "URL_PULL")
    {
        if (SOURCE) BODY["sources"] = [{ "url": SOURCE }]
    }

    // Send the request
    const RESPONSE = await fetch(`${prjConstants.ADMIN_API_URL}/liveInput`, {
        method: "PUT",
        headers: HEADERS,
        body: JSON.stringify(BODY)
    });

    // Check for success
    if (RESPONSE && RESPONSE.ok) {
        const JSON_RESPONSE = await RESPONSE.json();

        // Wait for the Live Input to be detached if it was just created
        await waitForLiveInputStatus(AUTH_TOKEN, JSON_RESPONSE.id, liveInputStatuses.Detached, 15, 1);

        return JSON_RESPONSE;
    }

    await apiExceptionHandler(RESPONSE, `Creating Live Input ${NAME} failed`);
}
